### ISSUE TITLE
Clarification about "working with any ref" changes

### ISSUE THREAD

**Issue by prince-chrismc** [EndBug/add-and-commit#351]
👋 First off, thank you for this action (it's a huge help)

I was updating my repo to the latest version [v8](https://github.com/EndBug/add-and-commit/releases/tag/v8.0.0) because I wanted to take advantage of

> now work with whatever ref has been checked out, without pulling or switching branches by default.

In my workflow on a `pull_request` event I simply use the default option with `actions/checkout@v2` [here](https://github.com/prince-chrismc/label-merge-conflicts-action/actions/runs/1727289637/workflow#L24)

However this fails https://github.com/prince-chrismc/label-merge-conflicts-action/runs/4892752781?check_suite_focus=true#step:8:79 with `fatal: You are not currently on a branch.` 

My use case is very similar to the example https://github.com/EndBug/add-and-commit#automated-linting just a different event 🤔 

Do you have any suggestion or ideas to help me debug this issue?

Thanks in advance ❤️

**Comment by EndBug** [comment]
Hi, thanks for using the action!

By default `actions/checkout` creates a detached head state when the run is triggered by a `pull_request` event. If you want to commit to the head branch you would need to checkout the branch the PR is coming from in the head repo.

I would set it up like this:

```yaml
- uses: actions/checkout@v2
    with:
      repository: ${{ github.event.pull_request.head.repo.full_name }}
      ref: ${{ github.event.pull_request.head.ref }}
```

You can find the full docs for payloads of `pull_request` events [here](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-32).

If you're planning on running this only on "internal" PRs then you could omit the `repository` input. If you're planning to use this with PRs coming from other forks, please keep in mind that you might not have write access to that repo. You can try setting up the repo with your PAT, but honestly, I've never tried. 

PS: I see that your workflow runs on both `pull_request` events and other ones: since this "custom checkout" is meant only for PRs, you could try having it run only for `pull_request` events, while other ones will trigger the usual checkout. If you wish to do so, you can use the `step.if` property, [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif)'s the docs.

Anyway, I think this is something worthy of putting in the FAQ section of the README, since other people might want to use the action in this scenario.

I hope this solves your issue!

**Comment by fasanosalvatore** [comment]
I have the same problem, the action fails (https://github.com/fasanosalvatore/FitDiary/runs/4908561536?check_suite_focus=true) on pull_request event even if I set the ref parameter as indicated.

**Comment by EndBug** [comment]
@fasanosalvatore `ref` is an input of `actions/checkout`, not of my action, you have to add it to [the first step of your workflow](https://github.com/fasanosalvatore/FitDiary/actions/runs/1733790040/workflow#L15).

**Comment by prince-chrismc** [comment]
> By default actions/checkout creates a detached head state when the run is triggered by a pull_request event.

was what I was missing, Thanks for the pointer.

I already handled forks and dependabot by [checking the permissions](https://github.com/prince-chrismc/check-actor-permissions-action) since that covers my use case better 😄 

[Just adding `ref`](https://github.com/prince-chrismc/label-merge-conflicts-action/pull/328/commits/f9b9496308543996c733983dd3d6fff29106e771) input to `actions/checkout` failed to push, doing some digging it seems the default token does not have workflow permission so I needed to use a PAT.


My PR is passing now.


ℹ️ I actually still have part of my CI running `v7` with `ref` and `token` for `actions/checkout` and it fails so thanks for this improvement!

**Comment by EndBug** [comment]
Happy to hear that! 
I'll close this issue as soon as I add the FAQ section to the README, let's keep this open in the meanwhile

**Comment by EndBug** [comment]
@all-contributors please add @prince-chrismc for their docs contribution

**Comment by EndBug** [comment]
This fix has been published in `v8.0.1` (also `v8` and `latest`). Thanks again for your contribution ❤️
